### ISSUE TITLE
Tabbing away from within the datepicker doesn't close the calendar

### ISSUE THREAD

**Issue by reicolee** [react-dates/react-dates#1467]
Is there any props where I can close/hide the calendar when I tab the focus away from within the calendar? 

Currently, I have to put an event listener on 'keydown' on the buttons inside the calendar, and trigger a setState to set the `focused` props to false when the key is `Tab`

**Comment by majapw** [comment]
@sumeet-jain recently added some functionality that would help with this I think! How are you rendering the calendar currently, @reicolee ?

**Comment by sebastiancaso** [comment]
@majapw Can you elaborate on the functionality that would help with this issue?  I just upgraded to 20.1.0 and it seems that I cannot tab out of the calendar once i've tabbed in.  If i reverse tab i get out of the calendar but it doesn't close.

**Comment by ccveer** [comment]
I am having the same issue.  It is reproducible here:
http://airbnb.io/react-dates/?path=/story/daterangepicker-drp--default

ADA accessibility compliance requires that the datepicker close when tabbing out of the input that renders it.

**Comment by ccveer** [comment]
When I fork the current repo, this bug is no longer occurring.  But the latest released version is broken and the story book link in the ReadMe is even more broken.  Also, when forking and running locally, there is a missing node-sass v^4.0.0 devDep.

**Comment by ljharb** [comment]
When you say the latest release is broken, do you mean, it has this bug, but it's fixed in the repo?

The storybook link should hopefully be fixed with #1673; as for node-sass, it's very sensitive to your node version, so locally i think it'll only work on node 12 or 14.

**Comment by ccveer** [comment]
> When you say the latest release is broken, do you mean, it has this bug, but it's fixed in the repo?
> 
> The storybook link should hopefully be fixed with #1673; as for node-sass, it's very sensitive to your node version, so locally i think it'll only work on node 12 or 14.

Good to know about node-sass.

Yes.  This bug is in the latest released version.  When I forked the repo yesterday though, the bug seems to be gone as it was working as expected.  I haven't tried an external usage but its working in storybook, which is making me think this bug will be fixed in the next release.  Speaking of which, do you know when that will be?  I've got a deadline I need to meet for our a11y grade.

**Comment by ljharb** [comment]
I don’t have a timeline, but I’m hoping for sooner rather than later.

**Comment by ljharb** [comment]
Closing, since it’s fixed in the default branch. Happy to reopen if the next release doesn’t solve it.

**Comment by AS-EBaiden** [comment]
How would one do it with the DayPickerSingleDateController and a custom Input

```
  const [focusedInput, setFocusedInput] = React.useState(true)
  const [date, setDate] = React.useState(null)
  const [visible, setVisible] = React.useState(false)
  const inputRef = React.useRef(null)

  const onDateChange = (newDate) => {
    if (newDate) {
      const momentDate = moment(newDate) // Convert to Moment object
      setDate(momentDate)
      setVisible(false)
    } else {
      setDate(null)
    }
  }
  const formattedDate = date ? date.format('MM/DD/YYYY') : ''

  const inputClick = (e) => {
    setVisible(true)
  }

  const handleBlur = (e) => {
    // Check if the blur event target is outside the input and DayPicker
    if (!inputRef.current.contains(e.relatedTarget)) {
      setVisible(false)
    }
  }
   return  <div>
        <CustomInput
          id="maskedInput"
          placeholder="mm/dd/yyyy"
          onClick={inputClick}
          onFocus={inputClick}
          onBlur={handleBlur} // Handle blur event
          value={formattedDate}
          {...props}
          ref={inputRef} // Attach the ref to the input element
        />

      {visible && (
        <div style={{ width: '300px', marginTop: '0.5em', position: 'absolute', zIndex: 99 }}>
          <DayPickerSingleDateController
            onDateChange={onDateChange}
            onFocusChange={({ focused }) => setFocusedInput(focused)}
            focused={focusedInput}
            onOutsideClick={() => setVisible(false)}
            date={date}
            {...props}
          />
        </div>
      )}
    </div>

```


I tried onBlur, and it's not responding

**Comment by mokshpyla** [comment]
This issue is not fixed
